File: ByTranslator/app/src/main/java/tr/com/fkocak/bytranslator/api/RetrofitClient.java
package tr.com.fkocak.bytranslator.api;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import retrofit2.converter.scalars.ScalarsConverterFactory;

/**
 * Created by fatihkocak on 20.09.2018.
 */

public class RetrofitClient {
    private static Retrofit retrofit = null;

    public static Retrofit getClient(String baseURL){
        if(retrofit == null){
            retrofit = new Retrofit.Builder()
                    .baseUrl(baseURL)
                    .addConverterFactory(ScalarsConverterFactory.create())
                    .build();
        }
        return retrofit;
    }

}

File: KotlinRetrofit/app/src/main/java/tr/com/fkocak/kotlinretrofit/api/RetrofitClient.kt
package tr.com.fkocak.kotlinretrofit.api

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class RetrofitClient {
    companion object {
        val retrofit : Retrofit? = null
        fun getClient(baseURL : String): Retrofit = if(retrofit == null)
            Retrofit.Builder()
                .baseUrl(baseURL)
                .addConverterFactory(GsonConverterFactory.create())
                .build()
        else
            retrofit
        }
}